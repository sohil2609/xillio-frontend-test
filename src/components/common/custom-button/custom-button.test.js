import React from 'react';
import { shallow } from 'enzyme';
import CustomButton from './custom-button.component';

const setUp = (props = {}) => {
	const component = shallow(<CustomButton {...props} />);
	return component;
};

describe('Custom Button Component', () => {
	let component;
	let mockProps;
	beforeEach(() => {
		mockProps = {
			text: 'submit',
			type: 'submit',
			onClick: jest.fn()
		};
		component = setUp(mockProps);
	});

	it('Should call onClick once on click', () => {
		const wrapper = component.find('[data-test="customButton"]');
		wrapper.simulate('click');
		expect(mockProps.onClick.mock.calls.length).toBe(1);
	});

	it('Should test snapshot', () => {
		expect(component).toMatchSnapshot();
	});
});
