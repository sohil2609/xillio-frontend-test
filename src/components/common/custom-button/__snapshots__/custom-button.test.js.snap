// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Button Component Should test snapshot 1`] = `ShallowWrapper {}`;

exports[`Header Component Should test snapshot 1`] = `ShallowWrapper {}`;
